src/ol/layers/MapboxStyleLayer.js

```diff
@@ -125,6 +140,26 @@ class MapboxStyleLayer extends Layer {
       //     this.addDynamicFilters();
       //   }),
     );
+
+    // Listen to click events
+    this.singleClickRef = this.map.on('singleclick', (e) => {
+      if (!this.clickCallbacks.length) {
+        return;
+      }
+
+      this.getFeatureInfoAtCoordinate(e.coordinate)
+        .then((d) => this.callClickCallbacks(d.features, d.layer, d.coordinate))
+        .catch(() => this.callClickCallbacks([], this, e.coordinate));
```

Comment:
this single click functionnality is implemented in WMSLayer and Vector Layer so I think it's better to implement it in src/ol/Layer.js

Reply:
I have removed the onClick listener from several classes as it is already defined in the common Layer.